test/jackdaw/streams_test.clj

```diff
@@ -643,7 +644,177 @@
             (is (= 3 (count keyvals)))
             (is (= [1 1] (first keyvals)))
             (is (= [1 3] (second keyvals)))
-            (is (= [1 6] (nth keyvals 2)))))))))
+            (is (= [1 6] (nth keyvals 2))))))))
+
+  (testing "suppress records per time window"
+    (let [topic-a (mock/topic "topic-a")
+          topic-b (mock/topic "topic-b")
+          topic-c (mock/topic "topic-c")
+
+          ;; if we don't set the `grace` period of the `TimeWindows`, the
+          ;; default is used: 24h - window
+          window       (Duration/ofMillis 100)
+          grace        (Duration/ofMillis 1)
+          time-windows (-> window
+                           TimeWindows/of
+                           (.grace grace))]
+
+      (with-open [driver (mock/build-driver (fn [builder]
+                                              (-> builder
+                                                  (k/ktable topic-a)
+                                                  (k/to-kstream)
+                                                  (k/group-by-key)
+                                                  (k/window-by-time time-windows)
+                                                  (k/reduce + topic-b)
+                                                  (k/suppress {})
+                                                  (k/to-kstream)
+                                                  (k/map (fn [[k v]] [(.key k) v]))
+                                                  (k/to topic-c))))]
+
+        (let [publish (partial mock/publish driver topic-a)]
+
+          (publish 1 1 3)
+          (publish 2 1 4)
+          (publish 201 1 1)
+          (publish 202 1 2)
+          (publish 303 1 1)
+
+          (let [keyvals (mock/get-keyvals driver topic-c)]
+            (is (= 2 (count keyvals)))
+            (is (= [1 7] (first keyvals)))
+            (is (= [1 3] (second keyvals))))))))
+
+  (testing "suppress records with a bounded buffer"
+    (let [topic-a (mock/topic "topic-a")
+          topic-b (mock/topic "topic-b")
+          topic-c (mock/topic "topic-c")
+
+          ;; if we don't set the `grace` period of the `TimeWindows`, the
+          ;; default is used: 24h - window
```

Comment:
including this comment once seems sufficient to me. could get rid of this

Reply:
removed